### PR TITLE
Enable a common SpiFlashLib in Silicon/CommonSocPkg

### PR DESCRIPTION
**Pull request by myguitar**
This is to consolidate SpiFlashLib into a single common one. There are
Silicon specific part to access SPI controller, so the Silicon specific
part will be done in a separate PchSpiLib.

For most of boards, common SpiFlashLib and PchSpiLib are added as below
- Silicon/CommonSocPkg/Library/SpiFlashLib/
- Silicon/CommonSocPkg/Library/PchSpiLib/

For APL, use common SpiFlashLib and PchSpiLib

For QEMU, use its own SpiFlashLib
- Silicon/QemuSocPkg/Library/SpiFlashLib/

For WHL(CFL), use common SpiFlashLib and its own PchSpiLib
- Silicon/CommonSocPkg/Library/SpiFlashLib/
- Silicon/CoffeelakePkg/Library/PchSpiLib/

Keep the existing BootloaderCommonPkg SpiFlashLib to avoid build failure
on WIP boards which are using BootloaderCommonPkg SpiFlashLib.
So, let's clean-up WIP boards and then remove the existing SpiFlashLib
in a separate PR.

Signed-off-by: Aiden Park <aiden.park@intel.com>

### CODE REVIEW

File: Silicon/CommonSocPkg/Library/SpiFlashLib/SpiFlashLib.c [hunk 1]
```diff
@@ -179,10 +179,22 @@ SpiConstructor (
...
+    DEBUG ((DEBUG_ERROR, "Failed to get SPI device info from Platform devices!\n"));
+    return EFI_NOT_FOUND;
+  }
+  DEBUG ((DEBUG_INFO, "PchSpiBase at 0x%08X\n", SpiInstance->PchSpiBase));
+
+  ///
+  /// Only for a device which has PCI bus/device/func info
+  ///
+  if (!(SpiInstance->PchSpiBase & 0xFF000000)) {
   SpiInstance->PchSpiBase = TO_MM_PCI_ADDRESS (SpiInstance->PchSpiBase);
```
Comment (mauricema) [inline Silicon/CommonSocPkg/Library/SpiFlashLib/SpiFlashLib.c]: align indent ?

File: Silicon/CoffeelakePkg/Library/PchSpiLib/PchSpiLib.c [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+**/
+VOID
+EFIAPI
+SetSpiBiosControlRegister (
+  IN  UINTN         PchSpiBase,
+  IN  UINT8         BiosCtlValue
+  )
+{
+  MmioAndThenOr8 (PchSpiBase + R_SPI_CFG_BC, (UINT8) ~B_SPI_CFG_BC_SRC, BiosCtlValue);
+}
```
Comment (mauricema) [inline Silicon/CoffeelakePkg/Library/PchSpiLib/PchSpiLib.c]: Fix  EOL ?

File: Silicon/CommonSocPkg/Library/SpiFlashLib/SpiFlashLib.c (hunk 1 above)
Comment (myguitar) [reply Silicon/CommonSocPkg/Library/SpiFlashLib/SpiFlashLib.c]: Updated in new push.

File: Silicon/CoffeelakePkg/Library/PchSpiLib/PchSpiLib.c (hunk 2 above)
Comment (myguitar) [reply Silicon/CoffeelakePkg/Library/PchSpiLib/PchSpiLib.c]: Updated in new push.

**Comment by mauricema** [comment]
Looks good to me.

**Pull request merged**